### PR TITLE
Pract04:task4 | Radoslav Peev

### PR DESCRIPTION
**Pull request by Radopeev**
Radoslav Peev
SI 3 7MI0600261

### CODE REVIEW

File: task4.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    if (source = nullptr) {
+        return;
+    }
+
+    char* ptr = destination;
+    
+    while (source != '\0') {
+        *destination = *source;
+        destination++;
+        source++;
```
Comment (YavorAlexandrov) [inline task4.cpp:30]: what if source's size is bigger than destination's capacity?

File: task4.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    *destination = '\0';
+
+    ptr=destination;
```
Comment (YavorAlexandrov) [inline task4.cpp:35]: why do you need this?

File: task4.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    int failedLogins=0;
+public:
+    Secret() = default;
+
+    Secret(const char* task, const char* password) {
+        strCopy(this->task, task);
+        strCopy(this->password, password);
+    }
+
+    char* getTask(const char* pwd) {
```
Comment (YavorAlexandrov) [inline task4.cpp:51]: get methods always return constant views of the objects. In this case it is necessary to return a const char* so the array is immutable.

Comment (YavorAlexandrov) [inline task4.cpp:51]: methods that don't change the state of the current object must be annotated with const after the method signature

File: task4.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    void setTask(const char* newTask, const char* passwd) {
+        if (strCmp(this->password, passwd) == 0) {
+            strCopy(this->task, task);
+        }
+        this->failedLogins++;
+    }
+
+    int getLoginFail() {
```
Comment (YavorAlexandrov) [inline task4.cpp:72]: this method is also const

**Review by YavorAlexandrov — CHANGES_REQUESTED** [review]

File: task4.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            strCopy(this->password, newPassword);
+        }
+        this->failedLogins++;
+    }
+
+    void setTask(const char* newTask, const char* passwd) {
+        if (strCmp(this->password, passwd) == 0) {
+            strCopy(this->task, task);
+        }
+        this->failedLogins++;
```
Comment (GeorgiTerziev02) [inline task4.cpp:69]: else failedlogins++

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

**Comment by GeorgiTerziev02** [comment]
Please check the uploaded solution

**Pull request closed without merging**